bionic/flow.py

```diff
@@ -820,69 +896,75 @@ def __call__(self, func):
         """
 
         decoration.init_accumulator_if_not_set_on_func(func)
-        provider = decoration.pop_accumulator_from_func(func).provider
-
-        if provider.attrs.protocols is None:
-            provider = ProtocolUpdateProvider(provider, DEFAULT_PROTOCOL)
-        if provider.attrs.docs is None:
-            docs = [None] * len(provider.attrs.names)
-            provider = AttrUpdateProvider(provider, "docs", docs)
-        if provider.attrs._can_persist is None:
-            provider = AttrUpdateProvider(provider, "_can_persist", True)
-        if provider.attrs._can_memoize is None:
-            provider = AttrUpdateProvider(provider, "_can_memoize", True)
-        if provider.attrs.changes_per_run is None:
```

Comment:
You are missing this case. I think it works because if `changes_per_run` is not set (and is `None` by default), it's the same as `False` everywhere wherever Bionic is using it. 

I think this is another indicator that `changes_per_run` should be a part of entity defn.

Reply:
Good catch! Actually I think we should have some kind of `FunctionAttributes` object that holds `version`, `changes_per_run`, etc.